model-optimizer/mo/ops/convolution.py

```diff
@@ -262,3 +274,16 @@ def infer(node: Node):
         PermuteAttrs.set_permutation(node.in_node(weights_index), node, node.soft_get('get_weights_permute', None))
         PermuteInputs().set_input_permutation(
             node.in_node(weights_index), node, 'input:{}'.format(weights_index), 'transpose')
+
+    @staticmethod
+    def reverse_infer(node: Node):
+        input_shape = node.in_port(0).data.get_shape()
+        if input_shape is None:
+            shape = None
+            # TODO FIXME this is ugly solution based on various attributes which may not be set in some cases
+            for attr in ['dilation', 'stride', 'pad']:
+                if node.has_valid(attr):
+                    shape = undefined_shape_of_rank(len(node.soft_get(attr)))
```

Comment:
I think it should be: `2+len(node.soft_get(attr))` because filter is applied along spatial dimensions

Reply:
Nope. At this stage of the model conversion the attribute semantic is different. Take a look at the shape infer function of the convolution. By default these attributes are filled with the number of ones or zeroes equal to the input tensor rank (len(input_shape)).